### PR TITLE
remove unused macros

### PR DESCRIPTION
**Pull request by rabauke**
The static analyzer of my IDE complains about unused macros. Let's get rid of some of them.

### CODE REVIEW

File: src/views/view.h [hunk 1]
```diff
@@ -115,10 +115,6 @@ typedef struct dt_mouse_action_t
   gchar name[256];
 } dt_mouse_action_t;
 
-#define DT_VIEW_ALL                                                                              \
```
Comment (TurboGit) [inline src/views/view.h:118]: Sounds legitimate to keep. This could be used as mask for example.

File: src/common/mipmap_cache.c [hunk 2]
```diff
@@ -932,11 +932,6 @@ void dt_mipmap_cache_get_with_caller(
   }
 }
 
-void dt_mipmap_cache_write_get_with_caller(dt_mipmap_cache_t *cache, dt_mipmap_buffer_t *buf, const uint32_t imgid, const int mip, const char *file, int line)
```
Comment (TurboGit) [inline src/common/mipmap_cache.c:935]: This is part of an API (not a macro) I think we should keep it. Could well be needed and better keep the API than letting user's redefining it somewhere else.

File: src/common/mipmap_cache.h [hunk 3]
```diff
@@ -124,16 +124,6 @@ void dt_mipmap_cache_get_with_caller(
     const char *file,
     int line);
 
-// convenience function with fewer params
```
Comment (TurboGit) [inline src/common/mipmap_cache.h:127]: Likewise for the header part.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Mostly ok, except for some, see comments.

**Comment by rabauke** [comment]
@TurboGit: Changes applied as requested.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**